api/src/main/java/org/apache/iceberg/catalog/Namespace.java

```diff
@@ -29,6 +31,8 @@
 public class Namespace {
   private static final Namespace EMPTY_NAMESPACE = new Namespace(new String[] {});
   private static final Joiner DOT = Joiner.on('.');
+  private static final Predicate<String> CONTAINS_WHITESPACE_OR_NULL_BYTE =
+      Pattern.compile("\\p{Zs}|\0|\u0000", Pattern.UNICODE_CHARACTER_CLASS).asPredicate();
```

Comment:
Why disallow whitespace?

Reply:
I guess whitespace wouldn't make a difference if it was put in backticks? I can remove it and then just add the `\0` and `\u0000`.

Reply:
Removed the whitespace check